frontend/src/components/SearchBar.tsx

```diff
@@ -0,0 +1,83 @@
+import { useForm } from 'react-hook-form';
+import { z } from 'zod';
+import { Form, FormControl, FormField, FormItem } from './ui/form';
+import { Search } from 'lucide-react';
+import { Input } from './ui/input';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { Button } from './ui/button';
+
+const formSchema = z.object({
+  searchQuery: z.string({
+    required_error: 'Restaurant name is required',
+  }),
+});
+
+type Props = {
+  onSubmit: (formData: SearchForm) => void;
+  placeHolder: string;
+  onReset?: () => void;
+};
+
+export type SearchForm = z.infer<typeof formSchema>;
+
+const SearchBar = ({ onSubmit, placeHolder, onReset }: Props) => {
+  const form = useForm<SearchForm>({
+    resolver: zodResolver(formSchema),
+  });
+
+  function handleReset(): void {
+    form.reset({
+      searchQuery: '',
+    });
+
+    if (onReset) {
+      onReset();
+    }
+  }
+
+  return (
+    <Form {...form}>
+      <form
+        onSubmit={form.handleSubmit(onSubmit)}
+        className={`flex items-center gap-3 justify-between flex-row border-2 rounded-full p-3 ${
+          form.formState.errors.searchQuery && 'border-red-500'
+        }`}
+      >
+        <Search
+          strokeWidth={2.5}
+          size={30}
+          className="ml-1 text-orange-500 hidden md:block"
+        />
+        <FormField
+          control={form.control}
+          name="searchQuery"
+          render={({ field }) => (
+            <FormItem className="flex-1">
+              <FormControl>
+                <Input
+                  {...field}
+                  className="border-none shadow-none text-xl focus-visible:ring-0"
+                  placeholder={placeHolder}
+                />
+              </FormControl>
+            </FormItem>
+          )}
```

Comment:
<div>

<strong>Bito Code Review Agent Run</strong> <a href=https://github.com/omkardongre/Food-Ordering-App/pull/36#issuecomment-2405200770>#5dee01</a><span> - 10/10/2024, 02:13 pm</span>
<br>

<div><small>🔴</small> High importance</div>

<div id="suggestion">
<div id="issue">
<b>Issue:</b> The current implementation only provides a visual indication of form errors (red border). It lacks an accessible error message for screen readers and users who may not perceive color differences. A similar issue was also found in frontend/src/components/SearchBar.tsx (line 51-65).
</div>
<div id="fix">
<b>Fix:</b> Add an aria-live region to display error messages, and include error text below the input field. Use the FormMessage component from your UI library to display these errors.
</div>
<details>
<summary>
<b>Code suggestion</b>
</summary>
<div id="code">

```
 @@ -51,16 +51,19 @@
          <FormField
            control={form.control}
            name="searchQuery"
 -          render={({ field }) => (
 +          render={({ field, fieldState }) => (
              <FormItem className="flex-1">
                <FormControl>
                  <Input
                    {...field}
                    className="border-none shadow-none text-xl focus-visible:ring-0"
                    placeholder={placeHolder}
 +                  aria-invalid={fieldState.invalid}
 +                  aria-describedby={fieldState.invalid ? 'searchQuery-error' : undefined}
                  />
                </FormControl>
 +              {fieldState.error && (
 +                <FormMessage id="searchQuery-error">{fieldState.error.message}</FormMessage>
 +              )}
              </FormItem>
            )}
          />
```
</div>
</details>
</div>
<br>


</div>

Is this a valid issue, or was it incorrectly flagged by the Agent?
- [ ] it was incorrectly flagged